src/Application.h

```diff
@@ -0,0 +1,117 @@
+//Copyright (c) 2018 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef APPLICATION_H
+#define APPLICATION_H
+
+#include "Slice.h" //To store the current slice.
+
+namespace cura
+{
+class Communication;
+
+/*!
+ * A singleton class that serves as the starting point for all slicing.
+ *
+ * The application provides a starting point for the slicing engine. It
+ * maintains communication with other applications and uses that to schedule
+ * slices.
+ */
+class Application
```

Comment:
Done: dde9da1a5dab72fc85ee0324b715ed987a58caaf

You could not construct Application (due to the singleton) but you could copy them. It was not bad performance-wise because this class contains 3 pointers and a size_t, but for style this is better.